packages/core/saved-objects/core-saved-objects-utils-server/src/merge_migrations.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ary, isFunction, mergeWith } from 'lodash';
+import type {
+  SavedObjectMigrationContext,
+  SavedObjectMigration,
+  SavedObjectMigrationMap,
+  SavedObjectUnsanitizedDoc,
+} from '@kbn/core-saved-objects-server';
+
+/**
+ * Composes two migrations into a single migration.
+ * ```
+ * mergeSavedObjectMigrations(outer, inner) -> (doc, context) => outer(inner(doc, context), context) }
+ * ```
+ *
+ * If at least one of the migrations is not deferred, the composed migration will not be deferred.
```

Comment:
I don't fully understand how what this means.
Using the example given here: 
```
mergeSavedObjectMigrationMaps({ '1.2.3': f }, { '1.2.3': g }) -> { '1.2.3': (doc, context) => f(g(doc, context), context) }
```
If `f` is deferred and `g` is not, will both `f` and `g` execute immediately or is `f` still deferred?

If they both run immediately, what's the point of registering `f` as deferred?